### PR TITLE
(SERVER-3389): Upgraded beaker-hostgenerator gem to add support for debian-12-x86_64

### PR DESCRIPTION
**Pull request by span786**
Upgraded beaker-hostgenerator gem to add support for debian-12-x86_64 in beaker tests

### CODE REVIEW

**Review by steveax — APPROVED** [review]

File: Gemfile [hunk 1]
```diff
@@ -19,7 +19,7 @@ gem 'rake', :group => [:development, :test]
 group :test do
   gem 'rspec'
   gem 'beaker', *location_for(ENV['BEAKER_VERSION'] || '~> 4.11')
-  gem "beaker-hostgenerator", *location_for(ENV['BEAKER_HOSTGENERATOR_VERSION'] || "~> 1.1")
+  gem "beaker-hostgenerator", *location_for(ENV['BEAKER_HOSTGENERATOR_VERSION'] || "~> 2.4.0")
```
Comment (joshcooper) [inline Gemfile:22]: We generally release BHG with new features as `2.y.0`, so I think you want the minor version to float like:

```suggestion
  gem "beaker-hostgenerator", *location_for(ENV['BEAKER_HOSTGENERATOR_VERSION'] || "~> 2.4")
```

Comment (donoghuc) [reply Gemfile:22]: Same page... I just did a similar review in pdb https://github.com/puppetlabs/puppetdb/pull/4000#discussion_r1759147238

**Review by donoghuc — CHANGES_REQUESTED** [review]
https://github.com/puppetlabs/puppetserver/pull/2877/files#r1759068324

File: Gemfile (hunk 1 above)
Comment (span786) [reply Gemfile]: Updated the code to float the minor version

**Comment by span786** [comment]
This [PR](https://github.com/puppetlabs/puppetserver/pull/2879) contains the beaker-hostgenerator changes  for `7.x`. Right now `smoke acceptance test suite` is failing in `7.x` pipeline which is not allowing the merge up from `7.x` to `main`. I have raised a PR https://github.com/puppetlabs/puppetserver/pull/2883 to fix the pipeline.

**Pull request closed without merging**